Include/cpython/code.h

```diff
@@ -2,6 +2,8 @@
 #  error "this header file must not be included directly"
 #endif
 
+// XXX Much of the code here should move to pycore_code.h.
```

Comment:
What should be moved? Is this just about the contents of the `PyCodeObject` struct?

Reply:
IMO, we should move anything that folks shouldn't even think about if they aren't dealing with the internals.  I'd say nearly everything there falls in that category.  I'd especially make `PyCodeObject` opaque in the public API, like it is in the limited API.

Reply:
Hm, anybody looking at code objects from C code is probably dealing with internals... What would be legitimate reasons to look at some fields here? I guess Pyjion and PyDev?

Reply:
Yeah, agreed.  I'm not going to tackle this part right now though.  It's going to take at least some discussion on python-dev.